### PR TITLE
gh-98931: Improve error message when the user types 'import x from y' instead of 'from y import x'

### LINKED ISSUE #98931: Improve syntax error for invalid imports
Is quite common to type `import x from y` instead of `from y import x`. We can easily offer a better error message in this case.

* PR: gh-98932

* PR: gh-105985

* PR: gh-105991

### PR DESCRIPTION
**Pull request by pablogsal**
Closes: #98931

* Issue: gh-98931

### CODE REVIEW

File: Grammar/python.gram [hunk 1]
```diff
@@ -1230,6 +1233,10 @@ invalid_group:
         RAISE_SYNTAX_ERROR_KNOWN_LOCATION(a, "cannot use starred expression here") }
     | '(' a='**' expression ')' {
         RAISE_SYNTAX_ERROR_KNOWN_LOCATION(a, "cannot use double starred expression here") }
+invalid_import:
+    | a='import' dotted_name 'from' dotted_name {
+        RAISE_SYNTAX_ERROR_STARTING_FROM(a, "Did you meant to use 'from ... import ...' instead?") }
```
Comment (lysnikolaou) [inline Grammar/python.gram]: Ditto in tests and docs.
```suggestion
        RAISE_SYNTAX_ERROR_STARTING_FROM(a, "Did you mean to use 'from ... import ...' instead?") }
```

**Review by lysnikolaou — COMMENTED** [review]
Good one! Just one suggestion for the error message.

File: Grammar/python.gram (hunk 1 above)
Comment (pablogsal) [reply Grammar/python.gram]: Done! Thanks for the review :)

Comment (lysnikolaou) [reply Grammar/python.gram]: Of course! I think you missed the one in `whatsnew`?

**Comment by pablogsal** [comment]
Will fix the doctest tomorrow

File: Doc/whatsnew/3.12.rst [hunk 2]
```diff
@@ -70,6 +70,17 @@ Important deprecations, removals or restrictions:
 
 * :pep:`623`, Remove wstr from Unicode
 
+Improved Error Messages
+=======================
+
+* Improve the :exc:`SyntaxError` error message when the user types ``import x
+  from y`` instead of ``from y import x``. Contributed by Pablo Galindo in :gh:`98931`.
+
+    >>> import a.y.z from b.y.z
```
Comment (CharlieZhao95) [inline Doc/whatsnew/3.12.rst:79]:
```suggestion
    >>> import a.y.z from b.y.z
    Traceback (most recent call last):
```
Fix doctest. Does this change meet your expectations? I noticed that there are similar cases in unit tests, so I didn't omit the traceback stack here to show more exception information.

**Review by lysnikolaou — APPROVED** [review]
Go!

**Comment by pablogsal** [comment]
:rocket:

**Pull request merged**